kernel/comps/softirq/src/lib.rs

```diff
@@ -113,11 +116,44 @@ impl SoftIrqLine {
 /// A slice that stores the [`SoftIrqLine`]s, whose ID is equal to its offset in the slice.
 static LINES: Once<[SoftIrqLine; SoftIrqLine::NR_LINES as usize]> = Once::new();
 
+/// The maximum number of IRQ lines we track (256 should have covered most common hardware).
+const MAX_IRQ_LINES: usize = 256;
+
+/// Global interrupt counter.
+static INTERRUPT_COUNTER: Once<[PerCpuCounter; MAX_IRQ_LINES]> = Once::new();
+
+/// Returns the interrupt counters for all IRQ lines.
+pub fn irq_count() -> [usize; MAX_IRQ_LINES] {
+    core::array::from_fn(|i| INTERRUPT_COUNTER.get().unwrap()[i].get())
+}
+
+/// A per-CPU counter for softirq execution.
+static SOFTIRQ_COUNTER: Once<[PerCpuCounter; SoftIrqLine::NR_LINES as usize]> = Once::new();
+
+/// Returns the execution count for all softirq lines.
+pub fn softirq_count() -> [usize; SoftIrqLine::NR_LINES as usize] {
+    core::array::from_fn(|i| SOFTIRQ_COUNTER.get().unwrap()[i].get())
+}
+
+/// Returns the softirq counters for a specific CPU.
+pub fn softirq_count_on_cpu(cpuid: CpuId) -> [usize; SoftIrqLine::NR_LINES as usize] {
```

Comment:
To be more precise, name this function `collect_per_softirq_counts_on_cpu` or other names that correctly imply the complexity (O(1) vs O(M), where M is the number of softirq lines) and return value (singular vs plural) of the operation.